ratelimits/limiter.go

```diff
@@ -105,9 +199,13 @@ func (l *Limiter) Check(ctx context.Context, txn Transaction) (*Decision, error)
 		// First request from this client. No need to initialize the bucket
 		// because this is a check, not a spend. A TAT of "now" is equivalent to
 		// a full bucket.
-		return maybeSpend(l.clk, txn.limit, l.clk.Now(), txn.cost), nil
+		d := maybeSpend(l.clk, txn.limit, l.clk.Now(), txn.cost)
+		d.transaction = txn
```

Comment:
This feels like it could result in a simple copy-paste error leaving `d.transaction` nil. What if `maybeSpend` took the whole txn as input (instead of taking the limit and the cost separately), and it set `d.transaction` before returning it?